### PR TITLE
rtnr: passthrough audio by stream copy

### PR DESCRIPTION
**Pull request by MingJenTai**
This PR passthrough multi channels audio from microphone using stream copy outside RTNR, instead of using the bypass function provided by RTNR.

Signed-off-by: Ming Jen Tai <mingjen_tai@realtek.com>

### CODE REVIEW

**Comment by MingJenTai** [comment]
> Thanks for the PR. I am thinking that we could align the implementation to the one used on Redrix/Gimble. Please see the attached patch. The main difference is that cd->process_enable is checked under rtnr_copy() to choose the path (process or bypass), rather than making a new func rtnr_passthrough(). It would be more straightforward.
> 
> [0001-RTNR-Decouple-switch-enable-control-from-params-blob.zip](https://github.com/thesofproject/sof/files/8752610/0001-RTNR-Decouple-switch-enable-control-from-params-blob.zip)

Force-pushed a commit to align the implementation.

**Review by lgirdwood — APPROVED** [review]
@johnylin76 LGTM, but pls check it's aligned to Gimble etc.

**Comment by lgirdwood** [comment]
@johnylin76 @cujomalainey good for you ?

**Review by cujomalainey — APPROVED** [review]

File: src/audio/rtnr/rtnr.c [hunk 1]
```diff
@@ -530,66 +535,82 @@ static int rtnr_copy(struct comp_dev *dev)
...
+		comp_get_copy_limits_with_lock(source, sink, &cl);
 
+		buffer_stream_invalidate(source, cl.source_bytes);
+
+		audio_stream_copy(&source->stream, 0, &sink->stream, 0,
+				source->stream.channels * cl.frames);
+
+		buffer_stream_writeback(sink, cl.sink_bytes);
+		comp_update_buffer_consume(source, cl.source_bytes);
+		comp_update_buffer_produce(sink, cl.sink_bytes);
```
Comment (lyakh) [inline src/audio/rtnr/rtnr.c:612]: you could do
```
if (!cd->process_enable) {
	...
	return 0;
}
```
in the beginning and then leave the rest as it was or with very few changes, the delta would be much smaller then

**Comment by lgirdwood** [comment]
Rerun the CI, looks like a lot of the DUTs were down or being used elsewhere.

**Comment by lgirdwood** [comment]
SOFCI TEST

**Pull request merged**